### DISCUSSION TITLE
I can't handle event from smart-contract

### DISCUSSION THREAD

**Discussion by Extroutor** [wevm/wagmi#1463] [Q&A] [answered]
Hello. I can't handle event from smart-contract. I can write contract and after that the event create. But i cant handle it. I use whatchContractEvent 

`const event = watchContractEvent({address: contractAddress, abi: contractABI, eventName: 'CreateStream',listener((*needed vars*) =>{*something*})`

I can see event and transaction in logs. I need to get some values from event, but i can't. I get this:

![image](https://user-images.githubusercontent.com/58516932/206997353-d63c712a-dc05-475d-bbf5-31c68d7291fd.png)

**Comment by jxom** [comment, marked as answer]
You'll need to pass that listener callback as a second parameter to `watchContractEvent`.

https://wagmi.sh/core/actions/watchContractEvent#usage

Sorry, the docs were previously outdated/incorrect. They just recently got updated with the correct usage.

**Reply by Extroutor** [reply]
Thank you so much!
